modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -453,89 +553,95 @@ impl WalletClientModule {
                 AutocommitError::ClosureError { error, .. } => error,
             })?;
 
-        Ok((operation_id, address))
+        Ok((operation_id, address, tweak_idx))
     }
 
-    pub async fn subscribe_deposit_updates(
+    pub async fn find_tweak_idx_by_operation_id(
         &self,
         operation_id: OperationId,
-    ) -> anyhow::Result<UpdateStreamOrOutcome<DepositState>> {
-        let operation_log_entry = self
+    ) -> anyhow::Result<TweakIdx> {
+        Ok(self
             .client_ctx
-            .get_operation(operation_id)
+            .module_db()
+            .clone()
+            .begin_transaction_nc()
             .await
-            .with_context(|| anyhow!("Operation not found: {}", operation_id.fmt_short()))?;
-
-        if operation_log_entry.operation_module_kind() != WalletCommonInit::KIND.as_str() {
-            bail!("Operation is not a wallet operation");
-        }
-
-        let operation_meta = operation_log_entry.meta::<WalletOperationMeta>();
+            .find_by_prefix(&PegInTweakIndexPrefix)
+            .await
+            .filter(|(_k, v)| future::ready(v.operation_id == operation_id))
+            .next()
+            .await
+            .ok_or_else(|| anyhow::format_err!("OperationId not found"))?
+            .0
+             .0)
+    }
 
-        if !matches!(
-            operation_meta.variant,
-            WalletOperationMetaVariant::Deposit { .. }
-        ) {
-            bail!("Operation is not a deposit operation");
+    pub async fn get_claimed_pegins(
+        &self,
+        dbtx: &mut DatabaseTransaction<'_>,
+        tweak_idx: TweakIdx,
+    ) -> Vec<(
+        bitcoin::OutPoint,
+        TransactionId,
+        Vec<fedimint_core::OutPoint>,
+    )> {
+        let outpoints = dbtx
+            .get_value(&PegInTweakIndexKey(tweak_idx))
+            .await
+            .map(|v| v.claimed)
+            .unwrap_or_default();
+
+        let mut res = vec![];
+
+        for outpoint in outpoints {
+            let claimed_peg_in_data = dbtx
+                .get_value(&ClaimedPegInKey {
+                    peg_in_index: tweak_idx,
+                    btc_out_point: outpoint,
+                })
+                .await
+                .expect("Must have a corresponding claim record");
+            res.push((
+                outpoint,
+                claimed_peg_in_data.claim_txid,
+                claimed_peg_in_data.change,
+            ));
         }
 
-        let mut operation_stream = self.notifier.subscribe(operation_id).await;
```

Comment:
While internally, only tests were consuming the intermediate states, currently nothing in the code is attempting to re-create the same level of visibility into intermediate states. It can be done as a follow-up.